src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/PodTemplateStepExecution.java

```diff
@@ -175,17 +175,10 @@ protected void finished(StepContext context) throws Exception {
                 return;
             }
             if (cloud instanceof KubernetesCloud) {
-                LOGGER.log(Level.INFO, "Removing pod template and deleting pod {1} from cloud {0}",
+                LOGGER.log(Level.INFO, "Removing pod template {1} from cloud {0}",
                         new Object[] { cloud.name, podTemplate.getName() });
                 KubernetesCloud kubernetesCloud = (KubernetesCloud) cloud;
                 kubernetesCloud.removeDynamicTemplate(podTemplate);
-                KubernetesClient client = kubernetesCloud.connect();
-                Boolean deleted = client.pods().withName(podTemplate.getName()).delete();
-                if (!Boolean.TRUE.equals(deleted)) {
-                    LOGGER.log(Level.WARNING, "Failed to delete pod for agent {0}/{1}: not found",
-                            new String[] { client.getNamespace(), podTemplate.getName() });
-                    return;
-                }
```

Comment:
Do not delete the pod with the `finished` callback - this is taken care of by `KubernetesSlave._terminate()`